integration/test_backend_configs.py

```diff
@@ -0,0 +1,67 @@
+"""Tests that we can use all available backend configs."""
+
+from datetime import datetime
+
+from pytket.backends.backendresult import BackendResult
+from pytket.circuit import Circuit
+
+import qnexus as qnx
+
+CONFIGS_REQUIRE_NO_MEASURE = [qnx.AerUnitaryConfig]
+CONFIGS_NOT_TO_EXECUTE = [
+    # We won't execute for these configs as they run on quantum hardware
+    qnx.IBMQConfig
+]
+
+
+def test_basic_backend_config_usage(
+    _authenticated_nexus: None,
+    backend_config: qnx.BackendConfig,
+    qa_project_name: str,
+) -> None:
+    """Test basic functionality of supported BackendConfigs."""
+
+    project_ref = qnx.projects.get(name_like=qa_project_name)
+
+    my_circ = Circuit(2, 2).H(0).CX(0, 1)
+
+    if backend_config.__class__ in CONFIGS_REQUIRE_NO_MEASURE:
+        my_circ.measure_all()
+
+    my_circ = qnx.circuits.upload(
+        circuit=my_circ,
+        name=f"QA BackendConfig circuit {backend_config.__class__.__name__}_{datetime.now()}",
+        description=f"This can be safely deleted. Test Run: {datetime.now()}",
+        project=project_ref,
+    )
+
+    compile_job_ref = qnx.start_compile_job(
+        programs=[my_circ],
+        name=f"QA BackendConfig compile job {backend_config.__class__.__name__}_{datetime.now()}",
+        optimisation_level=0,
```

Comment:
I think it is better to use an `optimisation_level` that reflects better the real-world usage, maybe `2`? Or do you think that's going to be make the compile job take too long?

Reply:
Oh no for sure we can use 2 - I don't think it will take too long